### PR TITLE
🐛 fix the AlreadyExists logic if the manifest is not cached by G…

### PR DESCRIPTION
**Pull request by elgnay**
…etter when applying a manifest

## Related issue(s)

Fixes #

### CODE REVIEW

File: pkg/common/apply/interface.go [hunk 1]
```diff
@@ -43,20 +45,24 @@ func Apply[T runtime.Object](
 	existing, err := getter.Get(requiredAccessor.GetName())
 	if errors.IsNotFound(err) {
 		actual, createErr := client.Create(ctx, required, metav1.CreateOptions{})
-		if errors.IsAlreadyExists(createErr) {
-			return required, false, nil
-		}
-		if createErr == nil {
+		switch {
+		case errors.IsAlreadyExists(createErr):
+			actual, getErr := client.Get(ctx, requiredAccessor.GetName(), metav1.GetOptions{})
```
Comment (qiujian16) [inline pkg/common/apply/interface.go:50]: should we add an comment on why we need to get again?

**Comment by qiujian16** [comment]
/approve

only thing is I think we need to add a comment

File: pkg/common/apply/interface.go (hunk 1 above)
Comment (elgnay) [reply pkg/common/apply/interface.go:53]: Comment added

**Pull request merged**